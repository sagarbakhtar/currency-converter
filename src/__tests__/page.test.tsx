import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Page from "../app/page";

describe("Page", () => {
  it("renders homepage unchanged", () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });

  it("renders a from", () => {
    render(<Page />);

    const from = screen.getByText("FROM:");

    expect(from).toBeInTheDocument();
  });
});
