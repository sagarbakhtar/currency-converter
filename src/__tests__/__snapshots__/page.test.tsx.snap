// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Page renders homepage unchanged 1`] = `
<div>
  <div
    class="flex min-h-screen justify-center items-center bg-white"
  >
    <div
      class="flex gap-20 p-12 shadow-[0_0_10px_2px_rgba(0,0,0,0.25)] w-xl"
    >
      <div
        class="flex flex-grow flex-col gap-10"
      >
        <div
          class="flex flex-col gap-8"
        >
          <div
            class="flex gap-6 items-center"
          >
            <div
              class="relative flex flex-grow flex-col gap-2"
            >
              <label
                class="text-xs font-light text-gray-400"
              >
                FROM
                :
              </label>
              <div
                class="flex items-center border-b border-blue-200 cursor-pointer pb-1 justify-between"
              >
                <div
                  class="flex items-center gap-2"
                >
                  <svg
                    height="32"
                    viewBox="0 0 32 32"
                    width="32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="#112f95"
                      height="24"
                      rx="4"
                      ry="4"
                      width="30"
                      x="1"
                      y="4"
                    />
                    <path
                      d="M27,4H5c-2.209,0-4,1.791-4,4V24c0,2.209,1.791,4,4,4H27c2.209,0,4-1.791,4-4V8c0-2.209-1.791-4-4-4Zm3,20c0,1.654-1.346,3-3,3H5c-1.654,0-3-1.346-3-3V8c0-1.654,1.346-3,3-3H27c1.654,0,3,1.346,3,3V24Z"
                      opacity=".15"
                    />
                    <path
                      d="M16 8.167L15.745 8.951 14.921 8.951 15.588 9.435 15.333 10.219 16 9.735 16.667 10.219 16.412 9.435 17.079 8.951 16.255 8.951 16 8.167z"
                      fill="#f6cd46"
                    />
                    <path
                      d="M16.255 22.565L16 21.781 15.745 22.565 14.921 22.565 15.588 23.049 15.333 23.833 16 23.349 16.667 23.833 16.412 23.049 17.079 22.565 16.255 22.565z"
                      fill="#f6cd46"
                    />
                    <path
                      d="M9.193 16.542L9.86 17.026 9.605 16.242 10.272 15.758 9.448 15.758 9.193 14.974 8.938 15.758 8.114 15.758 8.781 16.242 8.526 17.026 9.193 16.542z"
                      fill="#f6cd46"
                    />
                    <path
                      d="M12.596 9.079L12.342 9.863 11.517 9.863 12.184 10.347 11.93 11.131 12.596 10.647 13.263 11.131 13.009 10.347 13.675 9.863 12.851 9.863 12.596 9.079z"
                      fill="#f6cd46"
                    />
                    <path
                      d="M10.105 11.57L9.85 12.354 9.026 12.354 9.693 12.839 9.438 13.623 10.105 13.138 10.772 13.623 10.517 12.839 11.184 12.354 10.36 12.354 10.105 11.57z"
                      fill="#f6cd46"
                    />
                    <path
                      d="M10.36 19.161L10.105 18.377 9.85 19.161 9.026 19.161 9.693 19.646 9.438 20.43 10.105 19.945 10.772 20.43 10.517 19.646 11.184 19.161 10.36 19.161z"
                      fill="#f6cd46"
                    />
                    <path
                      d="M12.851 21.653L12.596 20.869 12.342 21.653 11.517 21.653 12.184 22.137 11.93 22.921 12.596 22.437 13.263 22.921 13.009 22.137 13.675 21.653 12.851 21.653z"
                      fill="#f6cd46"
                    />
                    <path
                      d="M23.886 15.758L23.062 15.758 22.807 14.974 22.552 15.758 21.728 15.758 22.395 16.242 22.14 17.026 22.807 16.542 23.474 17.026 23.219 16.242 23.886 15.758z"
                      fill="#f6cd46"
                    />
                    <path
                      d="M19.404 9.079L19.149 9.863 18.325 9.863 18.991 10.347 18.737 11.131 19.404 10.647 20.07 11.131 19.816 10.347 20.483 9.863 19.658 9.863 19.404 9.079z"
                      fill="#f6cd46"
                    />
                    <path
                      d="M21.483 12.839L21.228 13.623 21.895 13.138 22.562 13.623 22.307 12.839 22.974 12.354 22.15 12.354 21.895 11.57 21.64 12.354 20.816 12.354 21.483 12.839z"
                      fill="#f6cd46"
                    />
                    <path
                      d="M22.15 19.161L21.895 18.377 21.64 19.161 20.816 19.161 21.483 19.646 21.228 20.43 21.895 19.945 22.562 20.43 22.307 19.646 22.974 19.161 22.15 19.161z"
                      fill="#f6cd46"
                    />
                    <path
                      d="M19.658 21.653L19.404 20.869 19.149 21.653 18.325 21.653 18.991 22.137 18.737 22.921 19.404 22.437 20.07 22.921 19.816 22.137 20.483 21.653 19.658 21.653z"
                      fill="#f6cd46"
                    />
                    <path
                      d="M27,5H5c-1.657,0-3,1.343-3,3v1c0-1.657,1.343-3,3-3H27c1.657,0,3,1.343,3,3v-1c0-1.657-1.343-3-3-3Z"
                      fill="#fff"
                      opacity=".2"
                    />
                  </svg>
                  <span
                    class="font-medium text-xl"
                  >
                    EUR
                  </span>
                </div>
                <span
                  class=" w-3 transition-transform"
                >
                  <svg
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                      fill="inherit"
                    />
                  </svg>
                </span>
              </div>
              <ul
                class="opacity-0 -z-10 absolute transition-opacity top-full w-full bg-white border border-t-0 border-gray-300 rounded-xs divide-y shadow-sm text-sm"
              >
                <li
                  class="p-2 border-gray-100 flex gap-2 items-center text-sm/4 cursor-pointer"
                >
                  <svg
                    height="24"
                    viewBox="0 0 32 32"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1,24c0,2.209,1.791,4,4,4H27c2.209,0,4-1.791,4-4V15H1v9Z"
                      fill="#cb2e40"
                    />
                    <path
                      d="M27,4H5c-2.209,0-4,1.791-4,4v8H31V8c0-2.209-1.791-4-4-4Z"
                      fill="#fff"
                    />
                    <path
                      d="M5,28H27c2.209,0,4-1.791,4-4V8c0-2.209-1.791-4-4-4H5c-2.209,0-4,1.791-4,4V24c0,2.209,1.791,4,4,4ZM2,8c0-1.654,1.346-3,3-3H27c1.654,0,3,1.346,3,3V24c0,1.654-1.346,3-3,3H5c-1.654,0-3-1.346-3-3V8Z"
                      opacity=".15"
                    />
                    <path
                      d="M27,5H5c-1.657,0-3,1.343-3,3v1c0-1.657,1.343-3,3-3H27c1.657,0,3,1.343,3,3v-1c0-1.657-1.343-3-3-3Z"
                      fill="#fff"
                      opacity=".2"
                    />
                  </svg>
                  PLN
                </li>
                <li
                  class="p-2 border-gray-100 flex gap-2 items-center text-sm/4 cursor-pointer"
                >
                  <svg
                    height="24"
                    viewBox="0 0 32 32"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M31,8c0-2.209-1.791-4-4-4H5c-2.209,0-4,1.791-4,4v9H31V8Z"
                      fill="#2455b2"
                    />
                    <path
                      d="M5,28H27c2.209,0,4-1.791,4-4v-8H1v8c0,2.209,1.791,4,4,4Z"
                      fill="#f9da49"
                    />
                    <path
                      d="M5,28H27c2.209,0,4-1.791,4-4V8c0-2.209-1.791-4-4-4H5c-2.209,0-4,1.791-4,4V24c0,2.209,1.791,4,4,4ZM2,8c0-1.654,1.346-3,3-3H27c1.654,0,3,1.346,3,3V24c0,1.654-1.346,3-3,3H5c-1.654,0-3-1.346-3-3V8Z"
                      opacity=".15"
                    />
                    <path
                      d="M27,5H5c-1.657,0-3,1.343-3,3v1c0-1.657,1.343-3,3-3H27c1.657,0,3,1.343,3,3v-1c0-1.657-1.343-3-3-3Z"
                      fill="#fff"
                      opacity=".2"
                    />
                  </svg>
                  UAH
                </li>
              </ul>
            </div>
            <div
              class="pt-5"
            >
              <button
                class="w-6 fill-sky-400 cursor-pointer transition-transform active:rotate-180"
              >
                <svg
                  class="transition-transform rotate-180"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M32 96l320 0 0-64c0-12.9 7.8-24.6 19.8-29.6s25.7-2.2 34.9 6.9l96 96c6 6 9.4 14.1 9.4 22.6s-3.4 16.6-9.4 22.6l-96 96c-9.2 9.2-22.9 11.9-34.9 6.9s-19.8-16.6-19.8-29.6l0-64L32 160c-17.7 0-32-14.3-32-32s14.3-32 32-32zM480 352c17.7 0 32 14.3 32 32s-14.3 32-32 32l-320 0 0 64c0 12.9-7.8 24.6-19.8 29.6s-25.7 2.2-34.9-6.9l-96-96c-6-6-9.4-14.1-9.4-22.6s3.4-16.6 9.4-22.6l96-96c9.2-9.2 22.9-11.9 34.9-6.9s19.8 16.6 19.8 29.6l0 64 320 0z"
                    fill="inherit"
                  />
                </svg>
              </button>
            </div>
            <div
              class="relative flex flex-grow flex-col gap-2"
            >
              <label
                class="text-xs font-light text-gray-400"
              >
                TO
                :
              </label>
              <div
                class="flex items-center border-b border-blue-200 cursor-pointer pb-1 justify-between"
              >
                <div
                  class="flex items-center gap-2"
                >
                  <svg
                    height="32"
                    viewBox="0 0 32 32"
                    width="32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="#071b65"
                      height="24"
                      rx="4"
                      ry="4"
                      width="30"
                      x="1"
                      y="4"
                    />
                    <path
                      d="M5.101,4h-.101c-1.981,0-3.615,1.444-3.933,3.334L26.899,28h.101c1.981,0,3.615-1.444,3.933-3.334L5.101,4Z"
                      fill="#fff"
                    />
                    <path
                      d="M22.25,19h-2.5l9.934,7.947c.387-.353,.704-.777,.929-1.257l-8.363-6.691Z"
                      fill="#b92932"
                    />
                    <path
                      d="M1.387,6.309l8.363,6.691h2.5L2.316,5.053c-.387,.353-.704,.777-.929,1.257Z"
                      fill="#b92932"
                    />
                    <path
                      d="M5,28h.101L30.933,7.334c-.318-1.891-1.952-3.334-3.933-3.334h-.101L1.067,24.666c.318,1.891,1.952,3.334,3.933,3.334Z"
                      fill="#fff"
                    />
                    <rect
                      fill="#fff"
                      height="24"
                      width="6"
                      x="13"
                      y="4"
                    />
                    <rect
                      fill="#fff"
                      height="6"
                      width="30"
                      x="1"
                      y="13"
                    />
                    <rect
                      fill="#b92932"
                      height="24"
                      width="4"
                      x="14"
                      y="4"
                    />
                    <rect
                      fill="#b92932"
                      height="30"
                      transform="translate(32) rotate(90)"
                      width="4"
                      x="14"
                      y="1"
                    />
                    <path
                      d="M28.222,4.21l-9.222,7.376v1.414h.75l9.943-7.94c-.419-.384-.918-.671-1.471-.85Z"
                      fill="#b92932"
                    />
                    <path
                      d="M2.328,26.957c.414,.374,.904,.656,1.447,.832l9.225-7.38v-1.408h-.75L2.328,26.957Z"
                      fill="#b92932"
                    />
                    <path
                      d="M27,4H5c-2.209,0-4,1.791-4,4V24c0,2.209,1.791,4,4,4H27c2.209,0,4-1.791,4-4V8c0-2.209-1.791-4-4-4Zm3,20c0,1.654-1.346,3-3,3H5c-1.654,0-3-1.346-3-3V8c0-1.654,1.346-3,3-3H27c1.654,0,3,1.346,3,3V24Z"
                      opacity=".15"
                    />
                    <path
                      d="M27,5H5c-1.657,0-3,1.343-3,3v1c0-1.657,1.343-3,3-3H27c1.657,0,3,1.343,3,3v-1c0-1.657-1.343-3-3-3Z"
                      fill="#fff"
                      opacity=".2"
                    />
                  </svg>
                  <span
                    class="font-medium text-xl"
                  >
                    GBP
                  </span>
                </div>
                <span
                  class=" w-3 transition-transform"
                >
                  <svg
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                      fill="inherit"
                    />
                  </svg>
                </span>
              </div>
              <ul
                class="opacity-0 -z-10 absolute transition-opacity top-full w-full bg-white border border-t-0 border-gray-300 rounded-xs divide-y shadow-sm text-sm"
              >
                <li
                  class="p-2 border-gray-100 flex gap-2 items-center text-sm/4 cursor-pointer"
                >
                  <svg
                    height="24"
                    viewBox="0 0 32 32"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1,24c0,2.209,1.791,4,4,4H27c2.209,0,4-1.791,4-4V15H1v9Z"
                      fill="#cb2e40"
                    />
                    <path
                      d="M27,4H5c-2.209,0-4,1.791-4,4v8H31V8c0-2.209-1.791-4-4-4Z"
                      fill="#fff"
                    />
                    <path
                      d="M5,28H27c2.209,0,4-1.791,4-4V8c0-2.209-1.791-4-4-4H5c-2.209,0-4,1.791-4,4V24c0,2.209,1.791,4,4,4ZM2,8c0-1.654,1.346-3,3-3H27c1.654,0,3,1.346,3,3V24c0,1.654-1.346,3-3,3H5c-1.654,0-3-1.346-3-3V8Z"
                      opacity=".15"
                    />
                    <path
                      d="M27,5H5c-1.657,0-3,1.343-3,3v1c0-1.657,1.343-3,3-3H27c1.657,0,3,1.343,3,3v-1c0-1.657-1.343-3-3-3Z"
                      fill="#fff"
                      opacity=".2"
                    />
                  </svg>
                  PLN
                </li>
                <li
                  class="p-2 border-gray-100 flex gap-2 items-center text-sm/4 cursor-pointer"
                >
                  <svg
                    height="24"
                    viewBox="0 0 32 32"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M31,8c0-2.209-1.791-4-4-4H5c-2.209,0-4,1.791-4,4v9H31V8Z"
                      fill="#2455b2"
                    />
                    <path
                      d="M5,28H27c2.209,0,4-1.791,4-4v-8H1v8c0,2.209,1.791,4,4,4Z"
                      fill="#f9da49"
                    />
                    <path
                      d="M5,28H27c2.209,0,4-1.791,4-4V8c0-2.209-1.791-4-4-4H5c-2.209,0-4,1.791-4,4V24c0,2.209,1.791,4,4,4ZM2,8c0-1.654,1.346-3,3-3H27c1.654,0,3,1.346,3,3V24c0,1.654-1.346,3-3,3H5c-1.654,0-3-1.346-3-3V8Z"
                      opacity=".15"
                    />
                    <path
                      d="M27,5H5c-1.657,0-3,1.343-3,3v1c0-1.657,1.343-3,3-3H27c1.657,0,3,1.343,3,3v-1c0-1.657-1.343-3-3-3Z"
                      fill="#fff"
                      opacity=".2"
                    />
                  </svg>
                  UAH
                </li>
              </ul>
            </div>
          </div>
          <div
            class="flex gap-14"
          >
            <div
              class="flex flex-col gap-3 flex-grow"
            >
              <label
                class="text-xs font-light text-gray-400"
              >
                AMOUNT
                :
              </label>
              <div
                class="flex flex-col gap-1"
              >
                <div
                  class="flex flex-grow border-b border-blue-200 items-center"
                >
                  <div
                    class="flex flex-grow"
                  >
                    <input
                      class="outline-none flex w-full font-semibold text-2xl"
                      size="1"
                      type="number"
                      value="1.00"
                    />
                  </div>
                  <span
                    class="text-gray-500"
                  >
                    EUR
                  </span>
                </div>
                <p
                  class="h-4 text-sm/4 text-red-500"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-col gap-4"
        >
          <button
            class="flex items-center justify-center gap-2 w-full p-5 bg-green-400 rounded-xs text-white "
          >
            Convert
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
